script.go

```diff
@@ -275,6 +276,20 @@ func (p *Pipe) CountLines() (lines int, err error) {
 	return lines, p.Error()
 }
 
+// DecodeBase64 produces the string represented by the base64 encoded input.
+func (p *Pipe) DecodeBase64() *Pipe {
+	return p.Filter(func(r io.Reader, w io.Writer) error {
+		decoder := base64.NewDecoder(base64.StdEncoding, r)
```

Comment:
The use of `StdEncoding` reminds me that there are other encodings; not that common, perhaps, but they exist. It would be a shame if some user wanted to use `URLEncoding`, for example, but couldn't. What if there were a way to provide the encoding you want if it _isn't_ `StdEncoding`?

Reply:
@bitfield I believe we have a couple of options here: 
- Take a `*base64.Encoding` as input to the method. I believe this isn't preferred based on the conversation in the issue but let me know if we want to revisit that. 
- We can create a `WithBase64Encoding` method that can override which encoding is used

Happy to implement either approach! Let me know what you think.

Reply:
We _could_ use the trick of taking `...*base64.Encoding`, so that you don't have to pass an argument if you don't want to. I always think this is a bit too clever, though, and it has some puzzling edge cases (what if you pass more than one argument?)

Second-guessing myself again, I can't really imagine anyone wanting to use anything other than `StdEncoding`. Let's stick with that. If we _wanted_ to add an "optional" encoding argument later, it wouldn't break existing code (unless it referred to `p.EncodeBase64` as a function value, of course, but you can't win them all).